### PR TITLE
Frontend challenge/sami

### PR DESCRIPTION
**Pull request by samicodesit**

### CODE REVIEW

File: frontend-challenge-sami/src/store/index.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import Vue from "vue";
+import Vuex from "vuex";
+
+import { getProductPrice } from "@/utils";
+import { getProductWeight } from "@/utils";
```
Comment (ismail9k) [inline frontend-challenge-sami/src/store/index.js]: Importing a function from the same file.

File: frontend-challenge-sami/src/store/index.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    isEmptyCart: ({ cartItems }) => {
+      if (cartItems[0]) return false;
+
+      return true;
+    },
+    getCartPrice: ({ cartItems }) => {
+      const totalPrice = cartItems.reduce((total, item) => {
+        const productPrice = Number(getProductPrice(item.product));
+        const itemPrice = productPrice * item.quantity;
+        total += +itemPrice;
```
Comment (ismail9k) [inline frontend-challenge-sami/src/store/index.js]: Nice one 😉 
Although you already converted `pdouctPrice` to a number using `Number()` function.

File: frontend-challenge-sami/src/api/plugin.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  const headers = {};
+  if (token) {
+    headers.Authorization = `Bearer ${token}`;
+  }
+
+  return headers;
+}
+
+export function ApiPlugin(Vue) {
+  Vue.prototype.$api = callApi;
```
Comment (ismail9k) [inline frontend-challenge-sami/src/api/plugin.js]: 👍

File: frontend-challenge-sami/src/main.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import "./registerServiceWorker";
+import router from "./router";
+import store from "./store";
+import { ApiPlugin } from "./api/plugin";
+
+Vue.config.productionTip = false;
+
+Vue.use(ApiPlugin);
+
+Vue.filter("formatCurrency", value => {
```
Comment (ismail9k) [inline frontend-challenge-sami/src/main.js]: Note: Vue filters will deprecate in Vue 3

**Comment by samicodesit** [comment]
I switched from using the unary plus to Number() to favor readability, but I totally forgot to get rid of the operator.

File: frontend-challenge-sami/src/store/index.js (hunk 2 above)
Comment (samicodesit) [reply frontend-challenge-sami/src/store/index.js]: I switched from using the unary plus to Number() to favor readability, but I totally forgot to get rid of the operator.

**Pull request closed without merging**